### PR TITLE
Fix Multiclient E2E

### PR DESCRIPTION
**Pull request by nisdas**
**What type of PR is this?**

Bug Fix

**What does this PR do? Why is it needed?**

- [x] Following on from #11802 , we now correctly initialize lighthouse nodes to start with a pregenerated genesis state. 
- [x] Run multiclient api evaluator after epoch 0, to potentially handle an edge case correctly.
- [x] Fix finalized check to check our forkchoice store first, before hitting our db.

**Which issues(s) does this PR fix?**

N.A

**Other notes for review**

### CODE REVIEW

**Review by james-prysm — DISMISSED** [review]

File: testing/endtoend/components/lighthouse_validator.go [hunk 1]
```diff
@@ -324,46 +325,52 @@ func setupKeystores(valClientIdx, startIdx, numOfKeys int) (string, error) {
...
-		if err != nil {
-			return "", errors.Wrap(err, "could not marshal keystore to JSON file")
-		}
-		if err := file.WriteFile(fPath, encodedFile); err != nil {
-			return "", err
-		}
+	g, _ := errgroup.WithContext(context.Background())
+	for idx, pkey := range pubKeys {
+		pk := pkey
+		i := idx
```
Comment (kasey) [inline testing/endtoend/components/lighthouse_validator.go:331]: why not `for i, pk := range pubKeys`?

Comment (kasey) [reply testing/endtoend/components/lighthouse_validator.go:331]: oh i see, goroutine enclosure. although you don't need to copy the `idx` value, and you could do `pubKeyBytes := pk.Marshal()` outside the goroutine, which would eliminate the need to copy.

File: testing/endtoend/mainnet_scenario_e2e_test.go [hunk 2]
```diff
@@ -4,15 +4,16 @@ import (
 	"testing"
 
 	"github.com/prysmaticlabs/prysm/v3/config/params"
+	"github.com/prysmaticlabs/prysm/v3/runtime/version"
 	"github.com/prysmaticlabs/prysm/v3/testing/endtoend/types"
 )
 
 func TestEndToEnd_MainnetConfig_MultiClient(t *testing.T) {
-	e2eMainnet(t, false, true, params.E2EMainnetTestConfig().Copy(), types.WithValidatorCrossClient()).run()
+	e2eMainnet(t, false, true, types.StartAt(version.Phase0, params.E2EMainnetTestConfig()), types.WithValidatorCrossClient()).run()
```
Comment (kasey) [inline testing/endtoend/mainnet_scenario_e2e_test.go:12]: oops, this is my fault.

**Comment by kasey** [comment]
I'm not confident enough in my understanding of the attestation and fork choice changes to approve those, but the e2e changes all look good to me!

**Comment by rkapka** [comment]
One test fails

**Review by kasey — DISMISSED** [review]

**Review by kasey — APPROVED** [review]

**Pull request merged**